### PR TITLE
Propagate build_env variables from config map

### PR DESCRIPTION
**Pull request by chmeliik**
* CLOUDBLD-1228

Add variables to buildconfig, let openshift propagate them into
buildroot.

Signed-off-by: Adam Cmiel <acmiel@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: osbs/build/build_requestv2.py [hunk 1]
```diff
@@ -304,6 +304,14 @@ def set_data_from_reactor_config(self, reactor_config_data):
...
         self.set_registry_organization(reactor_config_data)
         self.set_required_secrets(reactor_config_data)
+        self.set_build_env(reactor_config_data)
+
+    def set_build_env(self, reactor_config_data):
+        """
+        Propagates build_env variables set in config map to build environment
+        """
+        build_env = reactor_config_data.get('build_env', [])
+        self.template['spec']['strategy']['customStrategy']['env'].extend(build_env)
```
Comment (MartinBasti) [inline osbs/build/build_requestv2.py:314]: :eyes: can we trust that configmap has correct format here?

**Comment by MartinBasti** [comment]
I'm thinking about blacklisting some names. like USER_PARAMS, BUILD, etc. We use for different purpose to avoid overwrites (or making sure that these won't be overwritten somehow else)

File: osbs/build/build_requestv2.py (hunk 1 above)
Comment (chmeliik) [reply osbs/build/build_requestv2.py:314]: Well it does get validated by schema, I guess I could write this in a way that is more likely to fail if we change the schema

**Comment by chmeliik** [comment]
> I'm thinking about blacklisting some names. like USER_PARAMS, BUILD, etc. We use for different purpose to avoid overwrites (or making sure that these won't be overwritten somehow else)

Couple changes:
- `set_build_env_vars()` will not set duplicate variables
- moved the method out of `set_data_from_reactor_config()`, because it is now important for it to run at the end of `render()`
- reverted old unit test and added a new one specifically for setting build_env_vars

File: osbs/build/build_requestv2.py (hunk 1 above)
Comment (MartinBasti) [reply osbs/build/build_requestv2.py]: It's too late for validation by atomic-rector :)

File: osbs/build/build_requestv2.py [hunk 2]
```diff
@@ -305,6 +310,26 @@ def set_data_from_reactor_config(self, reactor_config_data):
...
+        build_env_vars = reactor_config_data.get('build_env_vars', [])
+        for var in build_env_vars:
+            if var['name'] not in existing_vars:
+                # Could be just `env.append(var)`, but here we are using the reactor config
+                # without validating it (that happens later, in atomic-reactor). Do it this
+                # way to fail early if format does not match schema.
+                env.append({'name': var['name'], 'value': var['value']})
+                existing_vars.add(var['name'])
+                logger.info('Set environment variable from reactor config: %s', var['name'])
+            else:
```
Comment (MartinBasti) [inline osbs/build/build_requestv2.py:329]: I'd prefer fail, this warning will nobody notice and this shouldn't happen

Comment (chmeliik) [reply osbs/build/build_requestv2.py:329]: Good point, changed

File: osbs/build/build_requestv2.py (hunk 1 above)
Comment (chmeliik) [reply osbs/build/build_requestv2.py]: Yeah I realized, should we do some more thorough validation here? Or is it fine the way it is now?

Comment (MartinBasti) [reply osbs/build/build_requestv2.py]: I'd say it's fine now

**Comment by chmeliik** [comment]
Rebased, should fix py2 failure

**Comment by chmeliik** [comment]
Release notes: Environment variables can be defined in reactor config map, they will be propagated to build environment

**Pull request merged**